pandora-client-web/src/debug/observableRateLimit.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-inner-declarations */
+import { Observable } from '../observable';
+import { USER_DEBUG } from '../config/Environment';
+import { Assert, GetLogger } from 'pandora-common';
+
+if (USER_DEBUG) {
+	const OBSERVABLES = new WeakMap<object, {
+		lastEventTime: number;
+		storageEventCount: number;
+		async: boolean;
+	}>();
+
+	const OBSERVABLE_RATE_LIMITS = {
+		TIME_WINDOW: 1000,
+		MAX_EVENTS: 10,
+		ASYNC_DELAY: 10,
+	};
+
+	const desc = Object.getOwnPropertyDescriptor(Observable.prototype, 'value');
```

Comment:
Please... no... this kind of thing is nightmare to debug.
Alternative solution:
Make a class that extends the observable and from `observable` file do:
```ts
export const Observable = USER_DEBUG ? RateLimitedObservable : NormalObservable;
```